pkg/registry/core/pod/storage/eviction_test.go

```diff
@@ -31,10 +31,13 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/client-go/kubernetes/fake"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
 	podapi "k8s.io/kubernetes/pkg/api/pod"
 	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/apis/policy"
+	"k8s.io/kubernetes/pkg/features"
 )
 
 func TestEviction(t *testing.T) {
```

Comment:
We could also wrap all the tests with the feature flag to test both `disabled`/`enabled` and we could also test policies `""` and `PodReady` here. As mentioned above currently the eviction logic is a copy of the original one, so IMO it should not be that necessary. The price would be more complicated/unsightly tests. 

Any opinions on this?

Reply:
redoing the tests to cover all of the behaviour